includes/class-llms-form-templates.php

```diff
@@ -0,0 +1,680 @@
+				'options'        => array(),
+				'options_preset' => 'countries',
+				'required'       => ( 'required' === $option ),
+			)
+		);
+
+		$cols[] = $this->get_block(
+			'user-address-state',
+			array(
+				'field'          => 'select',
+				'id'             => 'llms_billing_state',
+				'label'          => __( 'State / Province', 'lifterlms' ),
+				'options'        => array(),
+				'options_preset' => 'states',
+				'required'       => ( 'required' === $option ),
+			)
+		);
+
+		$cols[] = $this->get_block(
+			'user-address-zip',
+			array(
+				'id'       => 'llms_billing_zip',
+				'label'    => __( 'Zip / Postal Code', 'lifterlms' ),
+				'required' => ( 'required' === $option ),
+			)
+		);
+
+		return $this->wrap_columns( $cols );
+
+	}
+
+	/**
+	 * Retrieve blocks for the street address row.
+	 *
+	 * Returns an empty array if address fields are disabled by legacy settings.
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	protected function get_row_address_street( $location ) {
+
+		$cols = array();
+
+		$option = get_option( sprintf( 'lifterlms_user_info_field_address_%s_visibility', $location ), 'required' );
+		if ( 'hidden' === $option ) {
+			return $cols;
+		}
+
+		$cols[] = array(
+			'content' => $this->get_block(
+				'user-address',
+				array(
+					'id'       => 'llms_billing_address_1',
+					'label'    => __( 'Street Address', 'lifterlms' ),
+					'required' => ( 'required' === $option ),
+				)
+			),
+			'width'   => 66.66,
+		);
+
+		$cols[] = array(
+			'content' => $this->get_block(
+				'user-address-additional',
+				array(
+					'id'               => 'llms_billing_address_2',
+					'placeholder'      => __( 'Apartment, suite, or unit', 'lifterlms' ),
+					'required'         => false,
+					'label_show_empty' => true,
+				)
+			),
+			'width'   => 33.33,
+		);
+
+		return $this->wrap_columns( $cols );
+
+	}
+
+	/**
+	 * Retrieve block(s) for the email address row.
+	 *
+	 * @since [version]
+	 *
+	 * @param string $location Form location. Accepts template options passed to `$this->get_template()`.
+	 * @return array
+	 */
+	protected function get_row_email( $location ) {
+
+		$cols = array();
+
+		$cols[] = $this->get_block(
+			'user-email',
+			array(
+				'field' => 'email',
+				'id'    => 'email_address',
+				'label' => __( 'Email Address', 'lifterlms' ),
+				'match' => 'email_address_confirm',
+			)
+		);
+
+		$option = get_option( sprintf( 'lifterlms_user_info_field_email_confirmation_%s_visibility', $location ) );
+		if ( ! llms_parse_bool( $option ) ) {
+			return $cols;
+		}
+
+		$cols[] = $this->get_block(
+			'user-email-confirm',
+			array(
+				'field'          => 'email',
+				'id'             => 'email_address_confirm',
+				'label'          => __( 'Confirm Email Address', 'lifterlms' ),
+				'match'          => 'email_address',
+				'data_store_key' => false,
+			)
+		);
+
+		return $this->wrap_columns( $cols, 'logged_out' );
```

Comment:
These fields are used in the dashboard edit account form as well, so they should be visible to all... right?